core/parachain/approval/store.hpp

```diff
@@ -0,0 +1,86 @@
+//
+// Created by iceseer on 11/7/22.
+//
+
+#ifndef KAGOME_STORE_HPP
+#define KAGOME_STORE_HPP
+
+#include "utils/non_copyable.hpp"
+
+namespace kagome::parachain {
+
+  template <typename K, typename V>
+  using StorePair = std::pair<std::decay_t<K>, std::decay_t<V>>;
+
+  template <typename T>
+  struct StoreUnit {
```

Comment:
Why do we need
```
Store<std::pair<int, A>, std::pair<int, B>> s;
as<std::pair<int, A>>(s).get(k);
```
instead of
```
std::map<int, A> a;
std::unordered_map<int, B> b;
get(a, k);
```

---

Maybe we can use helper free functions like `Option<V &> get(M<K, V>, K)` (this pr), or keep using raw iterators (`m[k]`, `m.find(k)`)